src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -36,6 +37,7 @@ private JsonDataSerializer()
                 ReferenceLoopHandling = ReferenceLoopHandling.Ignore
             };
 
+            messageSerializer = JsonSerializer.Create();
```

Comment:
Did you test if this solves #1757 ? According to the [docs ](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonSerializer.htm) .Create() uses the default settings, which I think caused the issue.

Reply:
@ObeA As mentioned by @singhsarab , `Create()` doesn't use default settings of JsonConvert. Check [docs](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonSerializer_Create.htm) here.